extension/partiql-extension-ion/src/boxed_ion.rs

```diff
@@ -413,13 +414,29 @@ impl<'a> RefTupleView<'a, Value> for BoxedIon {
             if let Ok(strct) = elt.expect_struct() {
                 for (k, elt) in strct {
                     if k.text().is_some_and(|k| matcher.matches(k)) {
+                        // TODO remove clone via ion-rs's new lazy element?
                         return Some(Cow::Owned(Self::new_value(elt.clone(), ctx.clone())));
                     }
                 }
             }
         }
         None
     }
+
+    fn tuple_fields_iter(&'a self) -> Box<dyn Iterator<Item = RefFieldView<'a, Value>> + 'a> {
+        let Self { doc, ctx } = self;
+        if let BoxedIonValue::Value(elt) = doc {
+            if let Some(strct) = elt.as_struct() {
+                return Box::new(strct.iter().map(|(name, value)| {
+                    let name = name.text();
+                    let value = Cow::Owned(Self::new_value(value.clone(), ctx.clone()));
+                    // TODO remove clone via ion-rs's new lazy element?
```

Comment:
can the clones be removed in this PR? otherwise can track in another issue

Reply:
No; It's unlcear whether/how they can at all. But if so, I would like to try https://github.com/amazon-ion/ion-rust/pull/918